sdk/python/kfp/client/client_test.py

```diff
@@ -41,5 +44,82 @@ def test_invalid(self, name: str):
             client.validate_pipeline_resource_name(name)
 
 
+PIPELINES_TEST_DATA_DIR = os.path.join(
+    os.path.abspath(os.path.join(__file__, os.path.pardir, os.path.pardir)),
+    'compiler', 'test_data', 'pipelines')
+
+PIPELINE_TEST_CASES = [
+    'pipeline_with_importer',
+    'pipeline_with_ontology',
+    'pipeline_with_if_placeholder',
+    'pipeline_with_concat_placeholder',
+    'pipeline_with_resource_spec',
+    'pipeline_with_various_io_types',
+    'pipeline_with_reused_component',
+    'pipeline_with_after',
+    'pipeline_with_condition',
+    'pipeline_with_nested_conditions',
+    'pipeline_with_nested_conditions_yaml',
+    'pipeline_with_loops',
+    'pipeline_with_nested_loops',
+    'pipeline_with_loops_and_conditions',
+    'pipeline_with_params_containing_format',
+    'lightweight_python_functions_pipeline',
+    'lightweight_python_functions_with_outputs',
+    'xgboost_sample_pipeline',
+    'pipeline_with_metrics_outputs',
+    'pipeline_with_exit_handler',
+    'pipeline_with_env',
+    'component_with_optional_inputs',
+    'pipeline_with_gcpc_types',
+    'pipeline_with_placeholders',
+    'pipeline_with_task_final_status',
+    'pipeline_with_task_final_status_yaml',
+    'component_with_pip_index_urls',
+]
+
+
+class TestOverrideCachingOptions(parameterized.TestCase):
+
+    @parameterized.parameters(PIPELINE_TEST_CASES)
+    def test_override_caching_from_yaml(self, pipeline_base_name: str):
+        pipeline_path = os.path.join(PIPELINES_TEST_DATA_DIR,
+                                     f'{pipeline_base_name}.yaml')
+        with open(pipeline_path) as f:
+            yaml_dict = yaml.safe_load(f)
+            test_client = client.Client(namespace='dummy_namespace')
+            test_client._override_caching_options(yaml_dict, False)
+            for _, task in yaml_dict['root']['dag']['tasks'].items():
+                assert task['cachingOptions']['enableCache'] == False
+
+    def test_override_caching_from_pipeline(self):
+        from kfp.v2.compiler import Compiler
```

Comment:
nit: import from main namespace and remove `.v2` part of path -- that only exists for backward compatibility